Week2/HYF-JS3-HW2/index.js

```diff
@@ -0,0 +1,124 @@
+  };
+
+  const mainContainer = createAndAppend('div', root, { id: 'container' });
+  const leftContainer = createAndAppend('div', mainContainer, { class: 'left-block frame' });
+  const table = createAndAppend('table', leftContainer);
+  const listContributor = createAndAppend('tbody', table);
+
+  const rightContainer = createAndAppend('div', mainContainer, { class: 'right-block frame' });
+  createAndAppend('p', rightContainer, {
+    text: 'Contributions',
+    class: 'contributor',
+  });
+
+  const list = createAndAppend('ul', rightContainer, { class: 'list1' });
+
+  function renderContributors(selRep) {
+    list.innerHTML = '';
+
+    for (let i = 0; i < selRep.length; i++) {
+      const li = createAndAppend('li', list, { class: 'contributor-item' });
+      const linkFor = createAndAppend('a', li, { target: '_blank', href: selRep[i].html_url });
+      createAndAppend('img', linkFor, { src: selRep[i].avatar_url, class: 'avatar', height: 48 });
+      const divCont = createAndAppend('div', linkFor, { class: 'contributor-data' });
+      createAndAppend('div', divCont, { text: selRep[i].login });
+      createAndAppend('div', divCont, {
+        text: selRep[i].contributions,
+        class: 'badge',
+      });
+    }
+  }
+
+  function renderRep(listContributors, repo) {
+    const content = [
+      { title: 'Repository', attribute: 'name' },
+      { title: 'Description', attribute: 'description' },
+      { title: 'Forks', attribute: 'forks' },
+      { title: 'Updated', attribute: 'updated_at' },
+    ];
+    listContributors.innerHTML = '';
+
+    for (let i = 0; i < content.length; i++) {
+      const headTitle = createAndAppend('tr', listContributors);
+      createAndAppend('td', headTitle, { text: `${content[i].title} :`, class: 'label' });
+      const cellContent = createAndAppend('td', headTitle);
+      if (content[i].attribute === 'name') {
+        createAndAppend('a', cellContent, {
+          href: repo.html_url,
+          text: repo.name,
+          target: '_blank',
+        });
+      } else {
+        cellContent.textContent = repo[content[i].attribute];
+      }
+    }
+    fetchJSON(repo.contributors_url).then(contributors => renderContributors(contributors));
+  }
+
+  function test(data) {
+    let optionArr = data;
+    optionArr = optionArr.sort((a, b) => a.name.localeCompare(b.name));
+
+    option(optionArr);
+
+    const selectedValue = document.getElementById('selector');
+    selectedValue.addEventListener('change', event => {
+      const selectedRepo = event.target.value;
+      renderRep(listContributor, optionArr[selectedRepo]);
+    });
+
+    renderRep(listContributor, optionArr[0]);
+  }
+  function errorHandler(err) {
+    createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  }
+
+  const HYF_REPS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+
+  fetchJSON(HYF_REPS_URL)
+    .then(data => test(data))
+    .catch(error => errorHandler(error));
+}
```

Comment:
What happened to `function main()` and `window.onload` which was in the original code?

Reply:
I try to used the .then and .catch in the function main() but it did not work then I find it in that way.
Let me try it again...

Reply:
>... I did not understand this part
>"You also removed some code from the starter file for reasons unclear to me, e.g. function main(), window.onload."
> Please can you put more light on it.

Look at the original code:

https://github.com/HackYourFuture/JavaScript3/blob/9bd7d0d563cb847ee7be566d66d05f7f7b42ee3d/homework/index.js#L33